buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/TestClustersPlugin.java

```diff
@@ -202,59 +144,14 @@ public void afterExecute(Task task, TaskState state) {
                     }
                     // always unclaim the cluster, even if _this_ task is up-to-date, as others might not have been
                     // and caused the cluster to start.
-                    Collection<ElasticsearchCluster> clustersUsedByTask = ((TestClustersAware) task).getClusters();
-                    if (clustersUsedByTask.isEmpty()) {
-                        return;
-                    }
-                    logger.info("Clusters were used, stopping and releasing permits");
-                    final int permitsToRelease;
-                    if (state.getFailure() != null) {
-                        // If the task fails, and other tasks use this cluster, the other task will likely never be
-                        // executed at all, so we will never be called again to un-claim and terminate it.
-                        clustersUsedByTask.forEach(cluster -> stopCluster(cluster, true));
-                        permitsToRelease = clustersUsedByTask.stream()
-                            .map(ElasticsearchCluster::getNumberOfNodes)
-                            .reduce(Integer::sum).get();
-                    } else {
-                        clustersUsedByTask.forEach(
-                            cluster -> claimsInventory.put(cluster, claimsInventory.getOrDefault(cluster, 0) - 1)
-                        );
-                        List<ElasticsearchCluster> stoppingClusers = claimsInventory.entrySet().stream()
-                            .filter(entry -> entry.getValue() == 0)
-                            .filter(entry -> runningClusters.contains(entry.getKey()))
-                            .map(Map.Entry::getKey)
-                            .collect(Collectors.toList());
-                        stoppingClusers.forEach(cluster -> {
-                            stopCluster(cluster, false);
-                            runningClusters.remove(cluster);
-                        });
-                    }
+                    ((TestClustersAware) task).getClusters()
+                        .forEach(cluster -> registry.stopCluster(cluster, state.getFailure() == null));
```

Comment:
We are passing the wrong value for `taskFailed` here. When `getFailure() == null` it means the task was successful. We need to either invert all this or rename the `taskFailed` param on `stopCluster()` to `taskSucceeded` or something so this all makes sense.